### PR TITLE
Never put player inside a solid when moving to a new world

### LINKED ISSUE #3088: /portal command can teleport the player to non-air blocks, making them stuck inside

### PR DESCRIPTION
**Pull request by LogicParrot**
Closes #3088

### CODE REVIEW

File: src/Entities/Player.cpp [hunk 1]
```diff
@@ -1762,7 +1762,48 @@ bool cPlayer::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
...
+						int NewY = Rel.y;
+						if (NewY < 0)
+						{
+							NewY = 0;
+						}
+						while (NewY < cChunkDef::Height - 2)
+						{
+							// If we find a position with enough space for the player
+							if ((Chunk->GetBlock(Rel.x, NewY, Rel.z) == E_BLOCK_AIR) &&
+							(Chunk->GetBlock(Rel.x, NewY + 1, Rel.z) == E_BLOCK_AIR))
```
Comment (NiLSPACE) [inline src/Entities/Player.cpp]: Is this valid or should the line have one more indent?

Comment (LogicParrot) [inline src/Entities/Player.cpp]: Fixed.

**Comment by Schwertspize** [comment]
In my opinion moving to a new world should always teleport to the world's spawn if not explicitly stated otherwise (by the plugin)

**Comment by LogicParrot** [comment]
@Schwertspize This PR does not prevent moving to the world's spawn point.

If you are referring to the way the `/portal` command works, then I agree. But this would require changes to the `core` plugin and is a seperate issue.

If you are referring to the API's `MoveToWorld` function, then I also agree, and it should be done in a seperate PR.

**Comment by NiLSPACE** [comment]
I disagree with the `MoveToWorld` method. I think it should calculate the proper coordinates automatically unless stated otherwise.

**Comment by LogicParrot** [comment]
@NiLSPACE I may have been a bit vague. I meant this: `MoveToWorld` with a specified position vector should teleport you to where you ask it to. `MoveToWorld` without a specified position should teleport to the spawn point. Currently the latter teleports you to a new world while retaining the coordinates you had in the old world. Is there any disagreement here?

**Comment by NiLSPACE** [comment]
I hade the opposite in mind, teleport to calculated area when none is given, but after giving it a second thought I think your way makes more sense. Perhaps we should make a few static functions as well to calculate the proper coordinates for overworld <-> nether changes?

**Comment by LogicParrot** [comment]
What do you mean by "calculated area"?

**Comment by NiLSPACE** [comment]
For example when going to the overworld from the nether every block counts as 8.

When I'm in 4, 64, 9 in the nether it means I'm in 32, 64, 72 in the overworld.

**Comment by LogicParrot** [comment]
@Schwertspize Sorry, I misunderstood your initial comment.

I agree that #3109 makes this redundant. Closing.

**Comment by LogicParrot** [comment]
Merged with #3097

**Pull request closed without merging**